Server/routes/api/getBudgetDetails.js

```
@@ -0,0 +1,48 @@
+            const { data: selectUserBudgetDetails, error: selectUserBudgetDetailsError } = await supabase
+                .from("userBudgets")
+                .select("category, budget")
+                .eq("id", userID);
+
+            if (selectUserBudgetDetailsError){
+                return response.status(400).json({ message: `this error occured trying to get the users budget details: ${selectUserBudgetDetailsError}` })
+            };
+
+            if (selectUserBudgetDetails && selectUserBudgetDetails.length > 0){
+                const budgetObject = selectUserBudgetDetails.reduce((acc, item) => ({
+                    ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)